hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/MergeIntoHoodieTableCommand.scala

```diff
@@ -495,7 +496,7 @@ case class MergeIntoHoodieTableCommand(mergeInto: MergeIntoTable) extends Hoodie
         KEYGENERATOR_CLASS_NAME.key -> classOf[SqlKeyGenerator].getCanonicalName,
         SqlKeyGenerator.ORIGIN_KEYGEN_CLASS_NAME -> tableConfig.getKeyGeneratorClassName,
         HoodieSyncConfig.META_SYNC_ENABLED.key -> enableHive.toString,
-        HiveSyncConfigHolder.HIVE_SYNC_MODE.key -> hiveSyncConfig.getString(HiveSyncConfigHolder.HIVE_SYNC_MODE),
+        HiveSyncConfigHolder.HIVE_SYNC_MODE.key -> hiveSyncConfig.getStringOrDefault(HiveSyncConfigHolder.HIVE_SYNC_MODE, HiveSyncMode.HMS.name()),
```

Comment:
Should we just add the default value below inside the `HiveSyncConfigHolder` class?
```
public static final ConfigProperty<String> HIVE_SYNC_MODE = ConfigProperty
      .key("hoodie.datasource.hive_sync.mode")
      .noDefaultValue()
      .withDocumentation("Mode to choose for Hive ops. Valid values are hms, jdbc and hiveql.");
```

Reply:
cc @xushiyan

Reply:
> Should we just add the default value below inside the `HiveSyncConfigHolder` class?
> 
> ```
> public static final ConfigProperty<String> HIVE_SYNC_MODE = ConfigProperty
>       .key("hoodie.datasource.hive_sync.mode")
>       .noDefaultValue()
>       .withDocumentation("Mode to choose for Hive ops. Valid values are hms, jdbc and hiveql.");
> ```

Here is to modify the global default value, which may affect a lot. I am not sure whether it is reasonable for each module. This PR only wants to change the default value corresponding to Spark SQL